e2e/src/utils.ts

```diff
@@ -59,14 +59,19 @@ export const testAssetDirInternal = '/data/assets';
 export const tempDir = tmpdir();
 export const asBearerAuth = (accessToken: string) => ({ Authorization: `Bearer ${accessToken}` });
 export const asKeyAuth = (key: string) => ({ 'x-api-key': key });
-export const immichCli = async (args: string[]) => {
-  let _resolve: (value: CliResponse) => void;
-  const deferred = new Promise<CliResponse>((resolve) => (_resolve = resolve));
-  const _args = ['node_modules/.bin/immich', '-d', `/${tempDir}/immich/`, ...args];
-  const child = spawn('node', _args, {
-    stdio: 'pipe',
+export const immichCli = (args: string[]) =>
+  executeCommand({ command: 'node', args: ['node_modules/.bin/immich', '-d', `/${tempDir}/immich/`, ...args] });
+export const immichAdmin = (args: string[]) =>
+  executeCommand({
+    command: 'docker',
+    args: ['exec', '-i', 'immich-e2e-server', '/bin/bash', '-c', `immich-admin ${args.join(' ')}`],
   });
 
+const executeCommand = ({ command, args }: { command: string; args: string[] }) => {
```

Comment:
I'm wondering if we need an object here or if `(command: string, ...args: string[])` is equally good. I think I prefer the latter

Reply:
Yeah that's probably fine

Reply:
I kept args as `args: string[]`. I think `...args: string[]` is a bit of an anti-pattern. There is a limitation on the number of params you can pass to a function and maybe some other considerations. The `Promise.all( [list] )` used to be like that, but has migrated to a list instead of individual args.